templates/layout.html

```diff
@@ -110,25 +110,97 @@
+          text-decoration: none;
+          cursor: pointer;
+        }
+      </style>
+
+      <li class="nav-item dropdown dropdown-pull-right">
+        <a id="myBtn" class="nav-link dropdown-toggle" data-toggle="dropdown" href="#" role="button"
+          aria-haspopup="true" aria-expanded="false">
           <i class="fa fa-globe" aria-hidden="true"></i> <span class="caret"></span>
         </a>
-        <div class="dropdown-menu" aria-labelledby="dropdown01">
-          {%- set eng_active = this.alt == 'en' %}
-          <a class="dropdown-item {% if eng_active %}active{% endif %}" href="{{ '.'|url(alt='en') }}">
-          English
-          </a>
-
-          {%- for alt, alternative in config.ALTERNATIVES.items()|sort -%}
-          {%- set alt_active = alt == this.alt -%}
-          {%- if alt != config.primary_alternative -%}
-          <a class="dropdown-item {% if alt_active %}active{% endif %}" href="{{ '.'|url(alt=alt) }}">
-          {{ transbag('translate', alt, alternative.name.en.lower()) }}
-          </a>
-          {%- endif -%}
-          {%- endfor -%}
+        <!-- Creating Modal Window Div -->
+        <div id="myModal" class="modal">
+          <!-- Modal Content Div [Language List Shows] -->
+          <div class="modal-content">
+            <span class="close">&times;</span>
+            {%- set eng_active = this.alt == 'en' %}
+            <a class="dropdown-item {% if eng_active %}active{% endif %}" href="{{ '.'|url(alt='en') }}">
+              English
+            </a>
+            {%- for alt, alternative in config.ALTERNATIVES.items()|sort -%}
+            {%- set alt_active = alt == this.alt -%}
+            {%- if alt != config.primary_alternative -%}
+            <a class="dropdown-item {% if alt_active %}active{% endif %}" href="{{ '.'|url(alt=alt) }}">
+              {{ transbag('translate', alt, alternative.name.en.lower()) }}
+            </a>
+            {%- endif -%}
+            {%- endfor -%}
+            <div class="text-center mt-2">
+              <small>This modal designed by: <b><a style="text-decoration: none; color: rgb(38, 36, 36);" target="_blank" href="https://github.com/pythoniseasy-hub">Gagan Aggarwal,</a></b> </br> Email: <b>gaggarwal124@gmail.com</b></small>
+            </div>
+          </div>
         </div>
+        <!-- This JS added By Gagan Aggarwal for Language Menu for Modal Functionality Add -->
+        <script>
+          var modal = document.getElementById("myModal");
+          var btn = document.getElementById("myBtn");
+          var span = document.getElementsByClassName("close")[0];
+          btn.onclick = function () {
+            modal.style.display = "block";
+          }
+          span.onclick = function () {
+            modal.style.display = "none";
+          }
+          window.onclick = function (event) {
+            if (event.target == modal) {
+              modal.style.display = "none";
+            }
+          }
+        </script>
```

Comment:
I'd prefer to do this *without* JS. A Pure CSS implementation should be possible; see https://codepen.io/Idered/pen/vytkH for one example of a possible approach.